modules/RTC/TPCUtils.js

```diff
@@ -93,35 +90,49 @@ export class TPCUtils {
         const isScreenshare = localVideoTrack.getVideoType() === VideoType.DESKTOP;
         let scalabilityMode = this.codecSettings[codec].useKSVC
             ? VideoEncoderScalabilityMode.L3T3_KEY : VideoEncoderScalabilityMode.L3T3;
-        let maxBitrate = codecBitrates.high;
+        const qualityLevel = VIDEO_QUALITY_LEVELS.find(level => level.height === newHeight);
+        let maxBitrate;
 
         if (this._isScreenshareBitrateCapped(localVideoTrack)) {
             scalabilityMode = VideoEncoderScalabilityMode.L1T3;
             maxBitrate = desktopShareBitrate;
         } else if (localVideoTrack.getVideoType() === VideoType.DESKTOP) {
             maxBitrate = codecBitrates.ssHigh;
+        } else if (qualityLevel) {
+            maxBitrate = codecBitrates[qualityLevel.level];
         }
 
         const config = {
             active: newHeight > 0,
             maxBitrate,
             scalabilityMode,
-            scaleResolutionDownBy: HD_SCALE_FACTOR
+            scaleResolutionDownBy: this.l2ScaleFactor
         };
 
         if (newHeight >= height || newHeight === 0 || isScreenshare) {
```

Comment:
```suggestion
        if (newHeight >= height || !config.active || isScreenshare) {
```

Reply:
Thanks for the suggestion, I will fold in the `qualityLevel` check as well, can't believe I missed that 🤦‍♀️